### ISSUE TITLE
Re: Edits on Record Headings// Was: Roadmap for Ripple-Qewd

### ISSUE THREAD

**Issue by ZuzooVn** [RippleOSI/Ripple-Qewd#39]
Hi Ripple Team,

Thanks for your great job.
Do you have any roadmap for Ripple-Qewd?

Thanks.
Nam Vu from ThinkLabs-VN

**Comment by ZuzooVn** [comment]
Currently, we want to contribute our effort to develop the remaining unfinished function and fix the known bug.
It would be helpful if I could know your roadmap.

Thanks.
Nam Vu from ThinkLabs-VN

**Comment by tony-shannon** [comment]
thanks @ZuzooVn 
We are finishing up our websites and documentation just now.
See our latest site with documentation here.. it still needs work to do but hope you like the style.
http://docs-showcase.ripple.foundation/

We have work to do on;
- linking verification mechanism to our authentication mechanism
- adding tests, 
- adding FHIR-openEHR mappings so we are compliant with key FHIR API 
- ensuring ready for enterprise deployment/ cloud at scale etc

Welcome your ideas..
What do you mean when you say "we want to contribute our effort to develop the remaining unfinished function and fix the known bug."

**Comment by ZuzooVn** [comment]
Thanks @tony-shannon 

Currently, we are using your great Ripple open source for our project,  thus we are fixing some known bugs and develop new functions:
- Create new patient profile.
- Update patient information
- Support Internationalization (i18n).
- Implement some unfinished functions: 
 + Create new clinical notes
 + Edit Allergy information
 + Edit Contact information
 + Create new Problems / Diagnoses
 ....
We want to move Ripple to production.

Thanks.
Nam Vu from ThinkLabs-VN

**Comment by tony-shannon** [comment]
ok thanks Nam Vu

If you have found bugs please report them under issues (label as bug) along
with related pull requests to fix them.
The notes, allergy, contact, problems functions should all be working ok.

If you want to develop new function, then you are welcome to try..
If the new function follow similar patterns to the past they may be easy
pull requests to handle.
If you need help from the core team then let us know.
As you'll understand we can only accept pull requests that we think are
good enough to be reused by others (inc documentation etc).

We are all working to move Ripple into production :o)
Thanks for the interest, help and support

Tony

**Comment by ZuzooVn** [comment]
Thanks @tony-shannon 

>The notes, allergy, contact, problems functions should all be working ok.

Here's my quick note for the status of Ripple function.

<img width="632" alt="screen shot 2017-07-05 at 3 32 59 pm" src="https://user-images.githubusercontent.com/1429230/27856098-22858672-6197-11e7-975c-02723d13c354.png">

As you can see, all of the "Edit" functions are no longer working. 
Do you have any specific plan to fix them?

Btw, I am going to report some new bugs.

Regards,
Nam Vu from ThinkLabs-VN

**Comment by tony-shannon** [comment]
thanks @ZuzooVn 
Not sure what version of Ripple_Qewd you're using? Can you check the date/version please
Please upgrade to the latest version, as of last Friday 
See the package.json file, should be "version": "2.28.0",

Can you provide some more detail on typical error please?
Are you seeing that in browser console or server log?
etc

thanks
Tony

**Comment by ZuzooVn** [comment]
Hi @tony-shannon ,

>Not sure what version of Ripple_Qewd you're using?

Version 2.28

> Can you provide some more detail on typical error please?
> Are you seeing that in browser console or server log?

Ref: https://github.com/RippleOSI/Ripple-Qewd/issues/47

>Yes we have work to do on Edits, you may be able to help on that.

Do you have any specific milestone for Edits? 

Thanks,
Nam

**Comment by tony-shannon** [comment]
Sorry for delay Nam.
We will do the work on Edits as funding allows, not timetabled just now.

Would you like to try amending some eg Problems, Medications, Allergies?
We may be able to provide guidance on how to do that..

thanks
Tony

**Comment by ZuzooVn** [comment]
Hi Tony, 

TL;DR: Yes, Please guide us.

Regards,
Nam Vu from ThinkLabs-VN

**Comment by tony-shannon** [comment]
ok
I'll ask @robtweed to provide some ideas here

Rob, need a chat on this please

**Comment by ZuzooVn** [comment]
@tony-shannon @robtweed 

> We will do the work on Edits as funding allows, not timetabled just now.
Do you have any updated news?

> Would you like to try amending some eg Problems, Medications, Allergies?
> We may be able to provide guidance on how to do that..
I am waiting for your guidance

**Comment by tony-shannon** [comment]
Sorry for the delay @ZuzooVn 
We have been focussed on a key project..

I'll ask @robtweed to offer some guidance here.

Apologies for the delay in getting back to you

**Comment by ZuzooVn** [comment]
@tony-shannon 
Thanks for your comment.
Currently, I saw that your team are developing PulseTile-React. Is it the key project that you mentioned here?

Without your guidance, we couldn't amend, because we got stuck at the logic.

I really hope that we can amend some this month.

**Comment by tony-shannon** [comment]
Hi @ZuzooVn 
Yes, we are developing a React version of PulseTile as well, but the key project I mentioned is this one.. .ie we are supporting Leeds City in England as they set up a Person Held Record built with our open source tools, PulseTile, Qewd-Ripple & EtherCIS.
See news here; https://www.digitalhealth.net/2017/10/leeds-develops-person-held-record-locals/

Am told that progressing the Edit issue is  linked to this issue in EtherCIS which we have identified in recent days.
https://github.com/ethercis/ethercis/issues/67

Once this is done, the Edits should be reasonably straighforward to progress and we will be actioning some of those ourselves for this PHR project.

Can you confirm you are using our full stack (PulseTile + QEWD-Ripple + EtherCIS)
Can you confirm what Edits are a priority for you?

thanks,
Tony

**Comment by ZuzooVn** [comment]
Hi @tony-shannon 

> Person Held Record

Do you fork the PulseTile, Qewd-Ripple & EtherCIS source code and develop specific functions for Leeds City or just use all of the original source code here?

> Can you confirm you are using our full stack (PulseTile + QEWD-Ripple + EtherCIS)

Yes, Of course.

>Can you confirm what Edits are a priority for you?

Currently, we are finding the way to edit Allergies.

Thanks,
Nam Vu from ThinkLabs-VN

**Comment by tony-shannon** [comment]
Hi @ZuzooVn 
We will reuse all the OS code from our PulseTile/Qewd-Ripple & EtherCIS repos, though we expect this project will improve each of those further.

As it happens the PulseTile team are planning to enable Edits to Allergies soon too, so once the related EtherCIS issue has been fixed this should be done pretty quickly. 
Will try to update you here

T

**Comment by ZuzooVn** [comment]
@tony-shannon 

>We will reuse all the OS code from our PulseTile/Qewd-Ripple & EtherCIS repos, though we expect this project will improve each of those further.

Will you opensource Leeds PHR? or Will you show us the way you set up a Person Held Record step by step, thus we can do it by our self.

>As it happens the PulseTile team are planning to enable Edits to Allergies soon too, so once the related EtherCIS issue has been fixed this should be done pretty quickly.

" I will try to fix it before going to Japan..." said chevalleyc. 

Do you have estimated time of arrival?

Thanks

Nam Vu

**Comment by tony-shannon** [comment]
@ZuzooVn 
All improvements to the Ripple stack of PulseTile, Qewd-Ripple, EtherCIS from the Leeds PHR effort will be open sourced. 

The secure login approach to our stack built into Qewd-Ripple and assigns users as either PHR, IDCR users is the basis of the PHR setup. Suggest you try it out and document how far you get.. that way your documentation can help others.

No ETA for that bug just now, Christian will aim to fix as soon as he can.

Please  take a look at our recent update on project support here.
https://github.com/RippleOSI/Ripple-Qewd/blob/master/README.md#productproject-support

**Comment by ZuzooVn** [comment]
@tony-shannon Thank for your valuable information.

Thank you once again for all your effort.

**Comment by tony-shannon** [comment]
No problem @ZuzooVn 

BTW update just in from Christian on that ETA at https://github.com/ethercis/ethercis/issues/67#issuecomment-336910483
_Thanks for all the feedback. I checked these issues today (all more or less related) and could figure out the missing bits and bad assumptions. I don't think I can post a fix before leaving for Japan though, but this will be fixed week 30/10. Many thanks again_

Sorry for the delay in fixing
T

**Comment by tony-shannon** [comment]
@ZuzooVn 
FYI Edits are now supported for Allergies & other headings
